internal/store/store.go

```diff
@@ -89,235 +106,268 @@ func customCATransport(ca []byte) http.RoundTripper {
 	}
 }
 
-func skipVerifyTransport() http.RoundTripper {
-	tlsClientConfig := &tls.Config{InsecureSkipVerify: true} //nolint:gosec // this is optional for some users
-	return &http.Transport{
-		ResponseHeaderTimeout: responseHeaderTimeout,
-		TLSHandshakeTimeout:   tlsHandshakeTimeout,
-		DialContext: (&net.Dialer{
-			Timeout:   timeout,
-			KeepAlive: keepAlive,
-		}).DialContext,
-		MaxIdleConns:          maxIdleConns,
-		MaxIdleConnsPerHost:   maxIdleConnsPerHost,
-		Proxy:                 http.ProxyFromEnvironment,
-		IdleConnTimeout:       idleConnTimeout,
-		ExpectContinueTimeout: expectContinueTimeout,
-		TLSClientConfig:       tlsClientConfig,
+func (s *Store) httpClient(httpTransport http.RoundTripper) (*http.Client, error) {
+	if s.username == "" || s.password == "" {
+		client := http.DefaultClient
+		client.Transport = httpTransport
+		return client, nil
 	}
-}
-
-type Config interface {
-	Service() string
-	PublicAPIKey() string
-	PrivateAPIKey() string
-	OpsManagerURL() string
-	OpsManagerCACertificate() string
-	OpsManagerSkipVerify() string
-	OpsManagerVersionManifestURL() string
-}
-
-func authenticatedClient(c Config) (*http.Client, error) {
 	t := &digest.Transport{
-		Username: c.PublicAPIKey(),
-		Password: c.PrivateAPIKey(),
-	}
-	if caCertificate := c.OpsManagerCACertificate(); caCertificate != "" {
-		dat, err := ioutil.ReadFile(caCertificate)
-		if err != nil {
-			return nil, err
-		}
-		t.Transport = customCATransport(dat)
-	} else if skipVerify := c.OpsManagerSkipVerify(); skipVerify == yes {
-		t.Transport = skipVerifyTransport()
-	} else {
-		t.Transport = http.DefaultTransport
+		Username: s.username,
+		Password: s.password,
 	}
-
+	t.Transport = httpTransport
 	return t.Client()
 }
 
-func defaultClient(c Config) (*http.Client, error) {
-	client := http.DefaultClient
-	if caCertificate := c.OpsManagerCACertificate(); caCertificate != "" {
-		dat, err := ioutil.ReadFile(caCertificate)
+func (s *Store) transport() (*http.Transport, error) {
+	switch {
+	case s.caCertificate != "":
+		dat, err := ioutil.ReadFile(s.caCertificate)
 		if err != nil {
 			return nil, err
 		}
-		client.Transport = customCATransport(dat)
-	} else if skipVerify := c.OpsManagerSkipVerify(); skipVerify == yes {
-		client.Transport = skipVerifyTransport()
-	} else {
-		client.Transport = defaultTransport
+		return customCATransport(dat), nil
+	case s.skipVerify:
+		return skipVerifyTransport, nil
+	default:
+		return defaultTransport, nil
 	}
-
-	return client, nil
 }
 
-// New get the appropriate client for the profile/service selected
-func New(c Config) (*Store, error) {
-	s := new(Store)
-	s.service = c.Service()
+// Option configures a Store.
+type Option func(s *Store) error
```

Comment:
[q] where is the implementation of this?

Reply:
oh I see, This is one of its implementation

```
func withAtlasClient(client *http.Client) Option {
	return func(s *Store) error {
		opts := []atlas.ClientOpt{atlas.SetUserAgent(userAgent)}
		if s.baseURL != "" {
			opts = append(opts, atlas.SetBaseURL(s.baseURL))
		}
		c, err := atlas.New(client, opts...)
		if err != nil {
			return err
		}
		s.client = c
		return nil
	}
}
```

Reply:
[nit] maybe moving this close to where we define `Option` would make it easier to find

Reply:
they are as closue as they can be, you have `func Options(opts ...Option) Option ` just after this, parmas and return type
and then after `Options` you have `func Service(service string) Option` again return type